src/server/zset_family.cc

```diff
@@ -2030,6 +2030,52 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInter(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<SetOpArgs> op_args_res = ParseSetOpArgs(args, false);
+
+  if (!op_args_res) {
+    return HandleOpStatus(cntx, op_args_res);
+  }
+  const auto& op_args = *op_args_res;
+  if (op_args.num_keys == 0) {
+    return SendAtLeastOneKeyError(cntx);
+  }
+
+  vector<OpResult<ScoredMap>> maps(shard_set->size(), OpStatus::SKIPPED);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpInter(shard, t, "", op_args.agg_type, op_args.weights, false);
+    return maps[shard->shard_id()];
+  };
+
+  cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  ScoredMap result;
+  for (auto& op_res : maps) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res)
+      return (*cntx)->SendError(op_res.status());
+
+    if (op_res->empty())
+      break;
```

Comment:
As is, this is a bug (which I got you to implement, sorry!). Since we're now breaking and not doing intersection, it will stop now and will return any previously computed results.

So I suggest to do:

```cpp
if (op_res->empty()) {
  result.clear();
  break;
}
```

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2164/commits/4f6b503dae98024f222ac0581b131b44b1679804